server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/jsieve/DiscardActionTest.java

```diff
@@ -0,0 +1,84 @@
+ *   with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing,
+ *   software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied.  See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ *
+ */
+
+package org.apache.james.transport.mailets.jsieve;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.apache.mailet.base.MailAddressFixture;
+import org.apache.mailet.base.test.FakeMail;
+import org.junit.Test;
+
+public class DiscardActionTest {
+
+    @Test
+    public void removeRecipientShouldWorkWhenOnlyOneRecipient() throws Exception {
+        FakeMail mail = FakeMail.builder()
+            .recipients(MailAddressFixture.ANY_AT_JAMES)
+            .build();
+        ActionContext actionContext = mock(ActionContext.class);
+        when(actionContext.getRecipient()).thenReturn(MailAddressFixture.ANY_AT_JAMES);
+
+        DiscardAction.removeRecipient(mail, actionContext);
+
+        assertThat(mail.getRecipients()).isEmpty();
+    }
+
+    @Test
+    public void removeRecipientShouldNotThrowWhenRecipientIsAbsent() throws Exception {
+        FakeMail mail = FakeMail.builder()
+            .recipients(MailAddressFixture.ANY_AT_JAMES)
+            .build();
+        ActionContext actionContext = mock(ActionContext.class);
+        when(actionContext.getRecipient()).thenReturn(MailAddressFixture.OTHER_AT_JAMES);
+
+        DiscardAction.removeRecipient(mail, actionContext);
+
+        assertThat(mail.getRecipients()).containsOnly(MailAddressFixture.ANY_AT_JAMES);
+    }
+
+    @Test
+    public void removeRecipientShouldNotThrowWhenRecipientIsNull() throws Exception {
+        FakeMail mail = FakeMail.builder()
+            .recipients(MailAddressFixture.ANY_AT_JAMES)
+            .build();
+        ActionContext actionContext = mock(ActionContext.class);
+        when(actionContext.getRecipient()).thenReturn(null);
+
+        DiscardAction.removeRecipient(mail, actionContext);
+
+        assertThat(mail.getRecipients()).containsOnly(MailAddressFixture.ANY_AT_JAMES);
+    }
+
+    @Test
+    public void removeRecipientShouldRemoveOnlyTheConcernedRecipient() throws Exception {
+        FakeMail mail = FakeMail.builder()
+            .recipients(MailAddressFixture.ANY_AT_JAMES, MailAddressFixture.OTHER_AT_JAMES)
+            .build();
+        ActionContext actionContext = mock(ActionContext.class);
+        when(actionContext.getRecipient()).thenReturn(MailAddressFixture.ANY_AT_JAMES);
+
+        DiscardAction.removeRecipient(mail, actionContext);
+
+        assertThat(mail.getRecipients()).containsOnly(MailAddressFixture.OTHER_AT_JAMES);
+    }
```

Comment:
You have a test showing that removing a non existing recipient has no effect ;-)